### ISSUE TITLE
Problem when examining 'dict' variable in variable explorer (Spyder)

### ISSUE THREAD

**Issue by iemcmillen** [spyder-ide/spyder#20537]
### What steps will reproduce the problem?

The problem occurred when double-clicking the 'matrix_eigvalues' variable in the variable explorer after running the code with the "run file button(F5)".

### Traceback
```python-traceback
  File "C:\Users\Isaac\anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionsdelegate.py", line 159, in createEditor
    editor.setup(value, key, icon=self.parent().windowIcon(),
  File "C:\Users\Isaac\anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1308, in setup
    self.widget = CollectionsEditorWidget(self, self.data_copy,
  File "C:\Users\Isaac\anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1249, in __init__
    self.editor = CollectionsEditorTableView(self, data, readonly,
  File "C:\Users\Isaac\anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1115, in __init__
    self.source_model = CollectionsModelClass(self, data, title,
  File "C:\Users\Isaac\anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 143, in __init__
    self.set_data(data)
  File "C:\Users\Isaac\anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 173, in set_data
    self.keys = sorted(list(data.keys()))
  File "C:\Users\Isaac\anaconda3\lib\site-packages\sympy\core\relational.py", line 384, in __nonzero__
    raise TypeError("cannot determine truth value of Relational")
TypeError: cannot determine truth value of Relational
```

## Versions

* Spyder version: 4.1.4 
* Python version: 3.8.3
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.5.0 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.16.1 (OK)
jedi =0.17.1                   :  0.17.1 (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  1.1.0 (OK)
paramiko >=2.4.0               :  2.7.1 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.0 (OK)
pygments >=2.0                 :  2.6.1 (OK)
pylint >=1.0                   :  2.5.3 (OK)
pyls >=0.34.0;<1.0.0           :  0.34.1 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.7.2 (OK)
qtconsole >=4.6.0              :  4.7.5 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.1.2 (OK)
spyder_kernels >=1.9.2;<1.10.0 :  1.9.2 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  19.0.1 (OK)

# Optional:
cython >=0.21                  :  0.29.21 (OK)
matplotlib >=2.0.0             :  3.2.2 (OK)
numpy >=1.7                    :  1.18.5 (OK)
pandas >=0.13.1                :  1.0.5 (OK)
scipy >=0.17.0                 :  1.5.0 (OK)
sympy >=0.7.3                  :  1.6.1 (OK)
```

**Comment by ccordoba12** [comment]
Hey @iemcmillen, thanks for reporting. Please post a simple code example that generates the variable that you're trying to view through the variable explorer.

That will help us to reproduce this error on our side and understand how to fix it.

**Comment by iemcmillen** [comment]
> Hey @iemcmillen, thanks for reporting. Please post a simple code example that generates the variable that you're trying to view through the variable explorer.
> 
> That will help us to reproduce this error on our side and understand how to fix it.

`import sympy as sym; import mpmath; import sys; import numpy as np
sys.modules['sympy.mpmath'] = mpmath
sym.init_printing(use_unicode=True)

x_1,x_2,x_3,x_4,x_5,x_6,x_7 = sym.symbols('x_1 x_2 x_3 x_4 x_5 x_6 x_7')

matrix_37B1 = sym.Matrix([[0,x_5,x_6,x_7,0,0,0],[-1*x_5,0,0,x_6,0,0,0],[-1*x_6,0,0,-1*x_5,0,0,0],[-1*x_7,-1*x_6,x_5,0,0,0,0],[0,0,0,0,0,0,0],[0,0,0,0,0,0,0],[0,0,0,0,0,0,0]])

matrix_eigvalues = matrix_37B1.eigenvals()`

**Comment by ccordoba12** [comment]
Thanks for the reproducible example @iemcmillen. However, I don't get an error with our latest version available in Anaconda (5.3.3) and Sympy 1.11.1:

![image](https://user-images.githubusercontent.com/365293/220808053-d4379b1f-f0c1-4176-854b-56c62530afdb.png)

Please update by closing Spyder, opening the Anaconda Prompt and running there the commands mentioned in [our documentation](http://docs.spyder-ide.org/current/installation.html#updating-with-anaconda).

To update Sympy, you need to run

    conda install sympy=1.11

Let us know if it works for you.

**Comment by ccordoba12** [comment]
Closing due to lack of response.
